Fix/claim gas limit

vasin-peach opened the pull request:
- `gasLimit` เมื่อ claim reward บน mainnet เพิ่มเป็น 25m จาก 7m (รวมถึงข้อความแจ้งเตือน)
- `gasLimit` ของ staking, un-staking, governance ใช้ estimate
- `sendTransactionAsync` (sendTx) ถ้าไม่ได้ส่ง `gasLimit` ไปจะ estimate ให้เองอัตโนมัติ

vasin-peach committed:
fix(sdk): add config GAS_CLAIM_LIMIT

vasin-peach committed:
fix(sdk): claim reward use 35m gas limit

vasin-peach committed:
fix(sdk): change GAS_CLAIM_LIMIT from 35m to 25m

vasin-peach committed:
chore(staking.modal): update gas limit in warning msg

tuahear commented on packages/javascript-sdk/src/config.ts:
```diff
@@ -26,4 +26,5 @@ export interface IConfig {
 }
 
 export const GAS_LIMIT = "7000000";
```
GAS_LIMIT_DEFAULT = "10000000";

tuahear replied:
// used for staking and governance

tuahear commented on packages/javascript-sdk/src/config.ts:
```diff
@@ -26,4 +26,5 @@ export interface IConfig {
 }
 
 export const GAS_LIMIT = "7000000";
+export const GAS_CLAIM_LIMIT = "25000000"
```
GAS_LIMIT_CLAIM_REWARD = "25000000";

vasin-peach committed:
docs(readme): update change gas limit example

vasin-peach committed:
chore(sdk): change GAS_LIMIT to GAS_LIMIT_DEFAULT with 10m gas limit

vasin-peach committed:
fix(core): update deploy-preview script

tuahear commented on README.md:
````diff
@@ -92,7 +92,8 @@ export const VALIDATOR_WALLETS: Record<string, {name: string, image: string}> =
 ```javascript
-export const GAS_LIMIT = "7000000";
+export const GAS_LIMIT_DEFAULT = "10000000";
+export const GAS_LIMIT_CLAIM_REWARD = "25000000"
````
ชื่อตัวแปร readme ไม่ตรงกับของจริงใน config.ts

vasin-peach committed:
docs(readme): update variable

vasin-peach committed:
fix(*): update gas price & gas limit

vasin-peach committed:
fix(*): fix gas price problem

- `gasLimit` split to testnet, mainnet
- `sendTransactionAsync` will estimate `gasLimit` if not pass `gasLimit`
- staking, governance use estimate gas limit
- claim reward warning message depend on testnet, mainnet

Closes #14, #15

vasin-peach commented:
deploy preview testnet, mainnet แล้วค่ะ

vasin-peach committed:
chore(sdk): add gasPrice to undelegate
